### PR TITLE
CE template with revisions

### PR DESCRIPTION
**Pull request by Alan-Cha**

### CODE REVIEW

File: testdata/ce/go.tpl [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        - metric: built-in/p95.0
+          upperLimit: 1
+    # task 3: if SLOs are satisfied, do nothing
+    - if: SLOs()
+      run: echo "SLOs are satisfied"
+    # task 4: if SLOs are not satisfied, revert to the previous version
+    - if: not SLOs()
+      run: |
+        # Get the name of the second to last revision
+        penultimateRevisionName=$(kubectl get revisions \
```
Comment (Alan-Cha) [inline testdata/ce/go.tpl:59]: This extracts the second to last revision name given the application name via `{{ .CEAppName }}`.

File: testdata/ce/go.tpl [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+metadata:
+  name: {{ .Name }}-{{ $suffix }}
+stringData:
+  experiment: |
+    # task 1: generate HTTP requests for https://example.com
+    # collect Iter8's built-in latency and error-related metrics
+    - task: gen-load-and-collect-metrics
+      with:
+        versionInfo:
+        - url: https://iter8-ce-demo.fgw1ut94lpp.ca-tor.codeengine.appdomain.cloud
```
Comment (sriumcp) [inline testdata/ce/go.tpl]: Should the URL value be replaced with something like ...

```
{{ .CEAppName }}-{{ .CEAppNamespace }}.ca-tor.codeengine.appdomain.cloud
```

File: testdata/ce/go.tpl [hunk 3]
```diff
@@ -0,0 +1,67 @@
+{{- $suffix := randAlphaNum 5 | lower -}}
+apiVersion: codeengine.cloud.ibm.com/v1beta1
+kind: JobRun
+metadata:
+  name: {{ .Name }}-{{ $suffix }}
```
Comment (sriumcp) [inline testdata/ce/go.tpl]: Could we please get rid of `{{ .Name }}` and replace it with a fixed string, like `experiment-run`.

**Review by sriumcp — CHANGES_REQUESTED** [review]
Please see comments.

The idea of extracting penultimate version inside the experiment is SUPER!

File: testdata/ce/go.tpl (hunk 2 above)
Comment (Alan-Cha) [reply testdata/ce/go.tpl]: I wonder how useful templating this will be because the URL also contains the location `ca-tor`, right?

Comment (sriumcp) [reply testdata/ce/go.tpl]: What  is this `ca-tor`? Also, does it make sense to template the whole URL field?

Comment (sriumcp) [reply testdata/ce/go.tpl]:
```
- url: {{ .URL }}
```

Comment (Alan-Cha) [reply testdata/ce/go.tpl]: `ca-tor` is the cluster zone. It's Toronto, Canada. But yes, I think it makes more sense to template the entire URL field.

**Comment by kalantar** [comment]
@Alan-Cha @sriumcp is this PR still relevant? It is very old. It predates v0.9.

**Comment by sriumcp** [comment]
Closing since it is not relevant any longer.

**Pull request closed without merging**